src/uart.h

```diff
@@ -0,0 +1,150 @@
+#define APBUART5_IRQ 21
+
+/* Number of simulator ticks between flushing the UARTS.  */
+/* For good performance, keep above 1000.  */
+#define UART_FLUSH_TIME	  5000
+
+/* New uart defines.  */
+#define UART_TX_TIME	1000
+#define UART_RX_TIME	1000
+
+/* Status Register definition, taken from GR712RC documentation:
+    (0)       DR - Data ready: Indicates that new data is available in the receiver holding register.
+    (1)       TS - Transmitter shift register empty: Indicates that the transmitter shift register is empty.
+    (2)       TE - Transmitter FIFO empty: Indicates that the transmitter FIFO is empty.
+    (3)       BR - Break received: Indicates that a BREAK has been received.
+    (4)       OV - Overrun: Indicates that one or more character have been lost due to overrun.
+    (5)       PE - Parity error: Indicates that a parity error was detected.
+    (6)       FE - Framing error: Indicates that a framing error was detected.
+    (7)       TH - Transmitter FIFO half-full: Indicates that the FIFO is less than half-full.
+    (8)       RH - Receiver FIFO half-full: Indicates that at least half of the FIFO is holding data.
+    (9)       TF - Transmitter FIFO full: Indicates that the Transmitter FIFO is full.
+    (10)      RF - Receiver FIFO full: Indicates that the Receiver FIFO is full.
+    (20 - 25) TCNT - Transmitter FIFO count: Shows the number of data frames in the transmitter FIFO.
+    (26 - 31) RCNT - Receiver FIFO count: Shows the number of data frames in the receiver FIFO.
+*/
+typedef enum {APBUART_DR = 0, APBUART_TS, APBUART_TE, APBUART_BR, APBUART_OV, APBUART_PE, APBUART_FE, APBUART_TH, 
+              APBUART_RH, APBUART_TF, APBUART_RF, APBUART_TCNT, APBUART_RCNT} apbuart_status_register;
+
+/* Control Register definition, taken from GR712RC documentation:
+    (0)     RE - Receiver enable: If set, enables the receiver.
+    (1)     TE - Transmitter enable: If set, enables the transmitter.
+    (2)     RI - Receiver interrupt enable: If set, interrupts are generated when a frame is received.
+    (3)     TI - Transmitter interrupt enable: If set, interrupts are generated when a frame is transmitted.
+    (4)     PS - Parity select: Selects parity polarity (0 = even parity, 1 = odd parity) (when implemented).
+    (5)     PE - Parity enable: If set, enables parity generation and checking (when implemented).
+    (7)     LB - Loop back: If set, loop back mode will be enabled.
+    (9)     TF - Transmitter FIFO interrupt enable: When set, Transmitter FIFO level interrupts are enabled.
+    (10)    RF - Receiver FIFO interrupt enable: When set, Receiver FIFO level interrupts are enabled.
+    (11)    DB - FIFO debug mode enable: When set, it is possible to read and write the FIFO debug register.
+    (31)    FA - FIFOs available: Set to 1, read-only. Receiver and transmitter FIFOs are available.
+*/
+typedef enum {APBUART_RE = 0, APBUART_CTRL_TE, APBUART_RI, APBUART_TI, APBUART_PS, APBUART_CTRL_PE, APBUART_LB = 7,
```

Comment:
This is definition of flag offsets, not the register.

Reply:
ack

Reply:
But not changed...